src/utils.py

```diff
@@ -232,60 +232,82 @@ def wrap_atom_predicates(atoms: Collection[LiftedOrGroundAtom],
     return new_atoms
 
 
-def run_policy_on_task(
-    policy: Callable[[State], Action],
-    task: Task,
-    simulator: Callable[[State, Action], State],
-    predicates: Collection[Predicate],
-    max_steps: int,
-    make_video: bool = False,
-    render: Optional[Callable[[State, Task, Action], List[Image]]] = None,
-    annotate_traj_with_goal: bool = False,
+def run_policy_until(
+        policy: Callable[[State], Action],
+        simulator: Callable[[State, Action], State],
+        init_state: State,
+        termination_function: Callable[[State], Action],
+        max_num_steps: int,
 ) -> Tuple[LowLevelTrajectory, Video, bool]:
-    """Execute a policy on a task until goal or max steps.
-
-    Return the low-level trajectory (optionally annotated with the
-    goal), and a bool for whether the goal was satisfied at the end.
+    """Execute a policy from an initial state, using a simulator.
+    Terminates when any of these conditions hold:
+    (1) the termination_function returns True,
+    (2) the policy throws OptionPlanExhausted,
```

Comment:
it's sad that (2) needs to be included in this function, but alas, it's all because we have the option weirdness where terminal can be True on the first step and it still works :(